src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1064,12 +1068,9 @@ public void ProcessBlocks(Func<int, IEnumerable<(ChainedHeader, Block)>> blockPr
                         // A wallet ahead of consensus should be truncated.
                         ChainedHeader fork = this.WalletRepository.FindFork(walletName, this.ChainIndexer.Tip);
 
-                        if (fork?.HashBlock != this.ChainIndexer.Tip?.HashBlock)
```

Comment:
This is wrong. If consensus tip is below wallet height the fork will equal the consensus tip and then not execute the rewind as desired.

Reply:
I checked this out as well and yeah good spot!